### PR TITLE
feat: new did implementation

### PR DESCRIPTION
**Pull request by weichweich**
Companion PR for https://github.com/KILTprotocol/mashnet-node/pull/146

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by Diiaablo95** [comment]
@rflechtner I tried the types in polkadot-js apps and they are working as expected as in https://github.com/KILTprotocol/mashnet-node/pull/146. Can you please try them out on the SDK and approve the PR if they work? Thanks! 😄

**Comment by weichweich** [comment]
before this is merged I would also like to have this tested with the polkadot apps. For the mashnet the apps should still use the old types (`types8`), where as they should use the new types (`types9`) for the mashnet-dev or the parachain.

**Comment by weichweich** [comment]
> Definitions are missing for `DispatchResultWithPostInfo` & `LockedBalance`. Maybe we need to update polkadot.js? Unfortunately this is true for the hosted polkadot apps as well

if we publish a new release of this project, the polkadot apps will be updated when they update the dependencies. which is quite frequently AFAIK

**Comment by Diiaablo95** [comment]
Ok `AttestationDetails` is actually my bad. I renamed the struct to avoid conflicts with the pallet name. But we are not using `DelegationNodeId` anywhere anymore, as that has been renamed `DelegationNodeIdOf`.

**Comment by Diiaablo95** [comment]
I guess a temporary fix would be to add also `DispatchResultWithPostInfo` and `LockedBalance`?

**Comment by weichweich** [comment]
> and I found the reason for the delegationNodeId:
> 
> ```
>               {
>                 name: revoke_root,
>                 args: [
>                   {
>                     name: root_id,
>                     type: DelegationNodeId
>                   },
>                   {
>                     name: max_children,
>                     type: u32
>                   }
>                 ],
>                 documentation: [
>                    Revoke a delegation root.,
>                   ,
>                    Revoking a delegation root results in the whole trust hierarchy,
>                    being revoked. Nevertheless, revocation starts from the leave nodes,
>                    upwards, so if the operation ends prematurely because it runs out of,
>                    gas, the delegation state would be consisent as no child would,
>                    survive its parent. As a consequence, if the root node is revoked,,
>                    the whole trust hierarchy is to be considered revoked.,
>                   ,
>                    * origin: the identifier of the revoker,
>                    * root_id: the ID of the delegation root to revoke,
>                    * max_children: the maximum number of nodes descending from the root,
>                      to revoke as a consequence of the root revocation
>                 ]
>               },
> ```
> 
> -> https://github.com/KILTprotocol/mashnet-node/blob/c13e1115d617b84475e44c65d905ac3bde3b612a/pallets/delegation/src/lib.rs#L316

Yes and we have now both `DelegationNodeIdOf` and `DelegationNodeId` which is the same type.

**Comment by Diiaablo95** [comment]
I would use `DelegationNodeIdOf` everywhere.

**Review by rflechtner — CHANGES_REQUESTED** [review]
sorry, I actually do have a concern that I overlooked earlier: can we change every property to use snake case? That's what the sdk expects, as it's a js convention

**Comment by weichweich** [comment]
> sorry, I actually do have a concern that I overlooked earlier: can we change every property to use snake case? That's what the sdk expects, as it's a js convention

I thought that the custom types expect to have camelCase? That's why i converted everything to camelCase. 😂  😅

**Comment by rflechtner** [comment]
> > sorry, I actually do have a concern that I overlooked earlier: can we change every property to use snake case? That's what the sdk expects, as it's a js convention
> 
> I thought that the custom types expect to have camelCase? That's why i converted everything to camelCase. 😂 😅

that's what I mean :D we still have some snake case in here, I got confused...

File: src/index.ts
Comment (rflechtner) [inline src/index.ts:101]: here!

File: src/index.ts
Comment (rflechtner) [inline src/index.ts:103]: and here!

File: src/index.ts
Comment (rflechtner) [inline src/index.ts:113]: and here!

File: src/index.ts
Comment (rflechtner) [inline src/index.ts:118]: and here!

**Comment by weichweich** [comment]
ah yes! 😁 they need to get converted! 👍🏻

**Review by rflechtner — APPROVED** [review]

**Pull request merged**
